src/LinearOperators.jl

```diff
@@ -3,7 +3,7 @@ module LinearOperators
 
 using FastClosures, Printf, LinearAlgebra, SparseArrays
 
-export AbstractLinearOperator, LinearOperator,
+export AbstractLinearOperator, LinearOperator, LinearOperator!,
```

Comment:
I appreciate the notation, but isn't it going to suggest that the operator modifies its argument? In other words, the user might think that writing `A * x` will modify `x`.

Reply:
Maybe we can keep `PreallocatedLinearOperator`? A `LinearOperator` will be preallocated by default. So `LinearOperator!` or `PreallocatedLinearOperator` will be called rarely by ourself.

Reply:
Sorry Abel, It wasn't clear. I said that we can keep PreallocatedLinearOperator for the name of the operator (instead of `LinearOperator!`).
Because if LinearOperator preallocated by default, the user will rarely need to call `PreallocatedLinearOperator` and `LinearOperator!` function himself.
So it wasn't a problem if the name (of the new method/operator) was a bit long.

But I didn't think about silent overwriting at this moment...